tron/api/controller.py

```diff
@@ -197,3 +198,40 @@ def delete_config(self, name, content, config_hash):
 
     def get_namespaces(self):
         return self.config_manager.get_namespaces()
+
+
+class EventsController:
+    COMMANDS = {'publish', 'discard'}
+
+    def publish(self, event):
+        if not EventBus.instance:
+            return dict(error='EventBus disabled')
+
+        if EventBus.has_event(event):
+            msg = f"event {event} already published"
+            log.warning(msg)
+            return dict(response=msg)
+
+        if not EventBus.publish(event):
+            msg = f'could not publish {event}'
+            log.error(msg)
+            return dict(error=msg)
+
+        return dict(response='OK')
+
+    def discard(self, event):
+        if not EventBus.instance:
```

Comment:
Do you really want to check EventBus.instance twice, at here and eventbus.py?

Reply:
this check isn't necessary for discard to function correctly, but i want to respond with correct error message in case eventbus is disabled.